### PR TITLE
[RFR] [BC Break] Migrate to material-ui 1.0

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] upgrade dependency (easy peasy)
- [x] suffer
- [x] make the example compile without error
- [x] sweat
- [x] make the example display the posts list without error
- [x] suffer again
- [x] Make the layout work in desktop
- [x] Make the layout work in mobile
- [x] cry
- [x] make the list views work 
- [ ] ~~find an alternative to `<Responsive>`~~ (nothing better in mui)
- [x] make the login / logout work
- [x] make the edit views work 
- [x] make the show views work
- [x] make the delete views work
- [x] make the create views work
- [ ] ~~remove useless components (like `<Responsive>`)~~
- [x] work hard
- [x] Reimplement `<AutocompleteInput>`
- [ ] ~~Replace the style override system (`className` instead of `style`)~~(to be done in another PR)
- [ ] ~~Reimplement `<SelectArrayInput>`~~ (to be done in another PR)
- [x] make the unit tests pass
- [x] make the e2e tests pass
- [x] rewrite documentation
- [ ] ~~rewrite Theming documentation~~ (to be done in another PR)
- [x] regret
- [x] write upgrade doc (ha!)
- [ ] ~~update screenshots (e.g. date picker)~~ (to be done in another PR)

Time currently spent: at least 20 hours
Current status:

image

image

image

image

image

### CODE REVIEW

**Comment by marcw** [comment]
Congrats on making that change. When do you feel it will be merged?

**Comment by fzaninotto** [comment]
In a few weeks I guess

**Comment by Phocea** [comment]
Just a question, but did you follow the discussions on the styling standard for material-ui, as highlighted in their [doc](http://www.material-ui.com/#/customization/styles), and is aor (or ra) going to also follow the same changes?

**Comment by fzaninotto** [comment]
Oh yes I did, and if you look at the changes in this PR, you'll see that I started converting inline styles to classes. I expect to remove all inline styles from aor.

**Comment by Phocea** [comment]
Now  I understand your pain :)

**Comment by natrim** [comment]
just a note: i noticed that they dropped datepicker component in material-ui 1.0
so i suggest adding to the todo: 

- [ ] find/write new datepicker component

**Comment by fzaninotto** [comment]
No, we'll do as they do and fallback to native datepicker

**Comment by djhi** [comment]
About the autocomplete component, it might be worth checking https://github.com/paypal/downshift

**Comment by fzaninotto** [comment]
Switching to RFR. I understand that this PR is impossible to review, so I'll merge it myself within a day anyway.

**Comment by oliviertassinari** [comment]
@fzaninotto I have been reading most of this pull request. I'm impressed by the effort you have put into it! Your effort won't be in vain. I will use this PR in order to improve the migration guide we have!

File: UPGRADE.md [hunk 1]
```diff
@@ -102,6 +102,70 @@ module.exports = {
...
+  const d = new Date(match[1], parseInt(match[2], 10) - 1, match[3]);
+  if (isNaN(d)) return;
+  return d;
+};
+<DateInput source="isodate" format={dateFormatter} parse={dateParser} label="ISO date" />
+```
+
+## Removed `<DateInput>` `options` props
+
+Material-ui 1.0 doesn't provide a real data picker, so the `options` prop of the `<DateInput>` is no longer supported.
```
Comment (marcw) [inline UPGRADE.md]: typo: should be `date` instead of `data`

**Pull request merged**